src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CaseRetriever.java

```diff
@@ -17,19 +26,43 @@
     }
 
     public CaseDetails retrieve(String jurisdiction, String caseRef) {
-        CcdAuthenticator info = factory.createForJurisdiction(jurisdiction);
-        return retrieveCase(jurisdiction, caseRef, info);
-    }
+        // not including in try catch to fast fail the method
+        CcdAuthenticator authenticator = factory.createForJurisdiction(jurisdiction);
 
-    private CaseDetails retrieveCase(String jurisdiction, String caseRef, CcdAuthenticator authenticator) {
-        return coreCaseDataApi.readForCaseWorker(
-            authenticator.getUserToken(),
-            authenticator.getServiceToken(),
-            authenticator.userDetails.getId(),
-            jurisdiction,
-            CASE_TYPE_ID,
-            caseRef
-        );
+        try {
+            CaseDetails caseDetails = coreCaseDataApi.readForCaseWorker(
+                authenticator.getUserToken(),
+                authenticator.getServiceToken(),
+                authenticator.userDetails.getId(),
+                jurisdiction,
+                CASE_TYPE_ID,
+                caseRef
+            );
+
+            logCaseDetails(caseDetails);
+
+            return caseDetails;
+        } catch (FeignException exception) {
+            if (exception.status() == NOT_FOUND.value()) {
+                log.info("Case not found. Ref:{}, jurisdiction:{}", caseRef, jurisdiction);
+
+                return null;
+            } else {
+                throw new CcdClientException(exception.getMessage(), exception);
+            }
+        } catch (Throwable exception) {
```

Comment:
Why're we catching throwable? and do we need to catch anything other than FeignException here?